packages/aws-library/src/aws_library/s3/_client.py

```diff
@@ -424,7 +425,16 @@ async def copy_object(
                     bytes_transfered_cb, file_name=f"{dst_object_key}"
                 )
             }
+        # NOTE: boto3 copy function uses copy_object until 'multipart_threshold' is reached then switches to multipart copy
+        # copy_object does not provide any callbacks so we can't track progress so we need to ensure at least the completion
+        # of the object is tracked
         await self._client.copy(**copy_options)
+        if bytes_transfered_cb:
```

Comment:
```python
if bytes_transfered_cb and object_metadata is None:
  ...
```

Reply:
no, if object_metadata is none I get them from S3. I do not do that everytime.